src/OrchardCore/OrchardCore.Scripting.JavaScript/JavaScriptEngine.cs

```diff
@@ -32,37 +31,21 @@ public IScriptingScope CreateScope(IEnumerable<GlobalMethod> methods, IServicePr
 
         public object Evaluate(IScriptingScope scope, string script)
         {
-            if (scope == null)
+            static void ThrowInvalidScopeTypeException()
```

Comment:
Use `ArgumentException.ThrowIfNullOrEmpty()` instead

Reply:
Haven't heard about that one 😉  there's `ArgumentNullException.ThrowIfNull()` but that won't fit here as it's invalid type.

Reply:
Why it's invalid, FYI many exceptions now have static `Throw()` method, we already use them in quite different places in the code base

Reply:
Well the original code checks that it has to be `JavaScriptScope`, if not, it's `ArgumentException`, didn't change the behavior here.

Reply:
The condition will still there, just we need to use the shortcut intead of creating new `ArgumentException` instance

Reply:
I still don't understand...